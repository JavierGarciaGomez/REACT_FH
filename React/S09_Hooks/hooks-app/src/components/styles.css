body{
    padding: 70px;
}

.blockquote {
    display: flex;
    background-color: cyan;
}