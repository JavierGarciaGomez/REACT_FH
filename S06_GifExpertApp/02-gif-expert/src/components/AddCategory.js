// ..., 77
import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ addCatHandler }) => {
  const [inputValue, setinputValue] = useState("");
  const submitHandler = (event) => {
    event.preventDefault();
    if (inputValue.trim().length < 2) return;
    addCatHandler(inputValue);
    setinputValue("");
  };

  const changeHandler = (e) => {
    setinputValue(e.target.value);
  };

  return (
    <form onSubmit={submitHandler}>
      <h2>Add Category</h2>
      <input type="text" onChange={changeHandler} value={inputValue} />
      <button type="submit">Agregar</button>
    </form>
  );
};

AddCategory.propTypes = {
  addCatHandler: PropTypes.func.isRequired,
};
