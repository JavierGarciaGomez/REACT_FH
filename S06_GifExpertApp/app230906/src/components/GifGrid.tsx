import { GifGridItem } from "./";
import { useFetchGifs } from "../hooks/useFetchGifs";

type Props = { category: string };

export const GifGrid = ({ category }: Props) => {
  const { gifs, loading } = useFetchGifs(category);

  return (
    <div>
      <h3>{category}</h3>
      {loading ? (
        <h2>Loading...</h2>
      ) : (
        <div className="card-grid">
          {gifs.map((gif) => (
            <GifGridItem key={gif.id} {...gif} />
          ))}
        </div>
      )}
    </div>
  );
};
