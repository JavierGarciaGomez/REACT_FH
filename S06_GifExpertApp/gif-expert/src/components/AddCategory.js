// ..., 77
import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setinputValue] = useState("");

  const handleInputValue = (ev) => {
    setinputValue(ev.target.value);
  };

  // ..., 77
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setinputValue("");
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputValue}></input>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
