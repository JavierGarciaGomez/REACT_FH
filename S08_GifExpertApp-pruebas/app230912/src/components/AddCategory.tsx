import { useState } from "react";

interface AddCategoryProps {
  onAddCategory: (category: string) => void;
}

export const AddCategory: React.FC<AddCategoryProps> = ({ onAddCategory }) => {
  const [inputValue, setinputValue] = useState<string>("");

  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) =>
    setinputValue(event.target.value);

  const handleSubmit = (event: React.FormEvent) => {
    event.preventDefault();
    if (inputValue.length < 3) return;
    onAddCategory(inputValue);
    setinputValue("");
  };

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Search gifs"
        value={inputValue}
        onChange={handleInputChange}
      ></input>
      <button type="submit">Submit</button>
    </form>
  );
};
