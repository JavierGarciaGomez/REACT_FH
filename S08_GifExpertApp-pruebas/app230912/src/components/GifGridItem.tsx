import { Gif } from "../types/types";

type Props = Gif;

export const GifGridItem = ({ title, url }: Props) => (
  <div className="card">
    <img src={url} alt={title} />
    <p>{title}</p>
  </div>
);
