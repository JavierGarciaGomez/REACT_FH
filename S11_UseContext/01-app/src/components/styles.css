body{
    padding: 50px;
}

.blockquote {
    display: flex;
    background-color: cyan;
}

.list-group-item{
    display: flex;
    align-items: center;
    justify-content: space-between;
}

p{
    cursor: pointer;
    margin: 0px;
}

.completed{
    text-decoration: line-through;
}