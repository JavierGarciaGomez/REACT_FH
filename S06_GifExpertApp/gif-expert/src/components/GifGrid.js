// 73, 74, 75, 78
import React, {useState, useEffect} from "react";
import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {
  // 75
  // const [images, setImages] = useState([])

  


  // 78
  const {data: images, loading} = useFetchGifs(category);
  

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && <p className="animate__animated animate__flash">'Cargando....'</p>}
      <div className="card-grid">
        
      {/* 85 */}
      <ol>
        {
          images.map((image, index) => 
          <GifGridItem key={image.id} {...image}/>            
          )
        }
      </ol>
      
      
    </div>
    </>
  );
};


GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}