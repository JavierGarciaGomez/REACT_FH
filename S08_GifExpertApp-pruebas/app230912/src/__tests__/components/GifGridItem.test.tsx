import { describe, expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../components";

describe("GifGrid ", () => {
  const title = "Saitama";
  const url = "https://one-punch.com/saitama.jpg";
  const id = "anId";

  test("It should match the snapshot", () => {
    const { container } = render(
      <GifGridItem id={id} title={title} url={url} />
    );
    expect(container).toMatchSnapshot();
  });

  test("It should show the img with the url and alt", () => {
    render(<GifGridItem id={id} title={title} url={url} />);
    const { src, alt } = screen.getByRole("img") as HTMLImageElement;
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("It should show the component title", () => {
    render(<GifGridItem id={id} title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
