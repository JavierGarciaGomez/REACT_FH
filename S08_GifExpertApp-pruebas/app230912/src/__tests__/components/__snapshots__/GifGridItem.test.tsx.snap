// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifGrid  > It should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
